server/services/store/sqlstore/sqlstore.go

```diff
@@ -33,6 +33,14 @@ type SQLStore struct {
 // a cluster mutex.
 type MutexFactory func(name string) (*cluster.Mutex, error)
 
+type NotSupportedError struct {
+	msg string
+}
+
+func (pe NotSupportedError) Error() string {
+	return pe.msg
+}
+
```

Comment:
I believe these are already defined.

Reply:
Actually, it is defined but in the mattermostauthlayer, so I change it to expose that error as part of the store package, and use it in the sqlstore and in the mattermostauthlayer.